Mathlib/Order/BourbakiWitt.lean

```diff
@@ -0,0 +1,227 @@
+  cSup_mem : ∀ (c : NonemptyChain α), ↑c ⊆ s → cSup c ∈ s
+
+lemma ici_isAdmissible : IsAdmissible x f (Ici x) where
+  base_isLeast := ⟨le_refl x, fun _ h ↦ h⟩
+  image_self_sub_self := by
+    rintro _ ⟨y, hy, rfl⟩
+    exact le_trans hy (f.le_map _)
+  cSup_mem := by
+    intro c hc
+    have ⟨y, hy⟩ := c.Nonempty'
+    apply le_trans (hc hy) (le_cSup _ _ hy)
+
+/-- The bottom admissible set with base point `x` and inflationary function `f` -/
+abbrev bot (x : α) (f : Inflationary α) : Set α := ⋂₀ {s | IsAdmissible x f s}
+
+lemma mem_bot_iff {y : α} :
+    y ∈ bot x f ↔ ∀ s ∈ {s | IsAdmissible x f s}, y ∈ s := by
+  unfold bot
+  exact mem_sInter
+
+lemma bot_isAdmissible : IsAdmissible x f (bot x f) where
+  base_isLeast := by
+    constructor
+    · rw [mem_bot_iff]
+      exact fun _ h ↦ h.base_isLeast.1
+    · intro y hy
+      rw [mem_bot_iff] at hy
+      exact (hy (Ici x) ici_isAdmissible)
+  image_self_sub_self := by
+    rintro _ ⟨y, hy, rfl⟩
+    rw [mem_bot_iff]
+    rintro s hs
+    exact hs.image_self_sub_self ⟨y, ⟨mem_sInter.1 hy _ hs, rfl⟩⟩
+  cSup_mem := by
+    intro c hc
+    rw [mem_bot_iff]
+    intro s hs
+    exact hs.cSup_mem c (subset_trans hc (sInter_subset_of_mem hs))
+
+lemma sub_bot_eq_bot {s : Set α} (h : IsAdmissible x f s) (h' : s ⊆ bot x f) : s = bot x f :=
+  subset_antisymm h' (sInter_subset_of_mem h)
+
+/-- The definition of an extreme point used in [serge_lang_algebra] -/
+structure IsExtremePt (x : α) (f : Inflationary α) (y : α) : Prop where
+  /-- By defintion, an extreme point is a member of `bot` -/
+  mem_bot : y ∈ bot x f
+  /-- By defintion, an extreme point `y` satisfies `f z ≤ y` for all `z < y` -/
+  map_le {z : α} (h : z ∈ bot x f) (h' : z < y) : f z ≤ y
+
+lemma map_mem_bot {y : α} (h : y ∈ bot x f) : f y ∈ bot x f := by
+  apply bot_isAdmissible.image_self_sub_self
+  use y
+
+lemma eq_bot {y : α} (hy : IsExtremePt x f y) : {z ∈ bot x f | z ≤ y ∨ f y ≤ z} = bot x f := by
+  apply sub_bot_eq_bot _ (sep_subset _ _)
+  apply IsAdmissible.mk
+  · constructor
+    · exact ⟨bot_isAdmissible.base_isLeast.1, Or.inl (bot_isAdmissible.base_isLeast.2 hy.mem_bot)⟩
+    · exact fun y h ↦ bot_isAdmissible.base_isLeast.2 h.1
+  · rintro _ ⟨z, ⟨hz, (hzy | hyz)⟩, rfl⟩ <;>
+      refine ⟨map_mem_bot hz, ?_⟩
+    · rcases le_iff_lt_or_eq.1 hzy with (hzy | rfl)
+      · left; exact hy.map_le hz hzy
+      · right; exact le_refl _
+    · right; exact le_trans hyz (f.le_map z)
+  · intro c hc
+    refine ⟨bot_isAdmissible.cSup_mem _ (subset_trans hc (sep_subset _ _)), ?_⟩
+    · by_cases h : ∀ z ∈ c, z ≤ y
+      · left; apply cSup_le c y h
+      · push_neg at h
+        rcases h with ⟨z, hz, hzy⟩
+        obtain h' := Or.resolve_left (hc hz).2 hzy
+        right
+        apply le_trans h' (le_cSup _ _ hz)
+
+lemma setOf_isExtremePt_eq_bot : {y | IsExtremePt x f y} = bot x f := by
+  apply sub_bot_eq_bot _ (fun _ h ↦ h.mem_bot)
+  apply IsAdmissible.mk
```

Comment:
I would say "the set of extreme points is admissible" is an interesting result here worthy of its independent theorem.